book/writing-programs/cycle-tracking.md

```diff
@@ -56,69 +56,44 @@ fn main() {
 
 This will log the cycle count for `block name` and include it in the `ExecutionReport` in the `cycle_tracker` map.
 
-## Tracking Cycles with Tracing
+### Profiling a ZKVM program 
 
-The `cycle-tracker` annotation is a convenient way to track cycles for specific sections of code. However, sometimes it can also be useful to track what functions are taking the most cycles across the entire program, without having to annotate every function individually.
+Profiling a zkvm program is a good way to get an understanding of what is bottlenecking your program. Note only one program may be profiled at a time.
 
-First, we need to generate a trace file of the program counter at each cycle while the program is executing. This can be done by simply setting the `TRACE_FILE` environment variable with the path of the file you want to write the trace to. For example, you can run the following command in the `script` directory for any example program:
+To profile a program, you have to setup a script to execute the program.
```

Comment:
```suggestion
To profile a program, you simply need to:
1. Enable the profiling feature for `sp1-sdk` in `script/Cargo.toml`
2. Set the env variable `TRACE_FILE=trace.json` and then call `ProverClient::execute()` in your script.

If you're executing a larger program (>100M cycles), you should set `TRACE_SAMPLE_RATE` to reduce the size of the trace file. A sample rate of `1000` means that 1 in every 1000 VM cycles is sampled. By default, every cycle is sampled.
```

Makes it more immediately clear how to profile a program, and explains exactly how `TRACE_SAMPLE_RATE` works. We don't need to link to an example script because it's in many other doc pages.

side note: should we make default sample rate more than 1?